src/VisualStudio/Core/Def/DocumentOutline/DocumentSymbolDataViewModelSorter.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Globalization;
+using System.Windows.Data;
+using System.Windows.Markup;
+using Roslyn.Utilities;
+
+namespace Microsoft.VisualStudio.LanguageServices.DocumentOutline
+{
+    /// <summary>
+    /// Sorts immutable collections of <see cref="DocumentSymbolDataViewModel"/>s 
+    /// </summary>
+    internal class DocumentSymbolDataViewModelSorter : MarkupExtension, IMultiValueConverter
+    {
+        public static DocumentSymbolDataViewModelSorter Instance { get; } = new();
+
+        public ImmutableArray<DocumentSymbolDataViewModel> Sort(ImmutableArray<DocumentSymbolDataViewModel> items, SortOption sortOption)
+            => (ImmutableArray<DocumentSymbolDataViewModel>)Convert(new object[] { items, sortOption }, typeof(ImmutableArray<DocumentSymbolDataViewModel>), null, CultureInfo.CurrentCulture);
+
+        public object Convert(object[] values, Type targetType, object? parameter, CultureInfo culture)
```

Comment:
when does the strongly typed vs weakly typed versions get called?

Reply:
WPF calls the weakly-typed version. The strongly-typed version is used for testing, I'll add a comment explaining this.